src/index.d.ts

```diff
@@ -0,0 +1,608 @@
+	field?: number;
+	power?: number;
+
+}
+
+export class GradientMapMaterial extends MaterialBase<GradientMapMaterialDefines> {
+
+	constructor( parameters?: GradientMapMaterialParameters );
+
+	map: Texture | null;
+	minColor: Color;
+	minValue: number;
+	maxColor: Color;
+	maxValue: number;
+	field: number;
+	power: number;
+
+}
+
+export interface GraphMaterialParameters extends ShaderMaterialParameters {
+
+	dim?: boolean;
+	thickness?: number;
+	graphCount?: number;
+	graphDisplay?: Vector4;
+	overlay?: boolean;
+	xRange?: Vector2;
+	yRange?: Vector2;
+	colors?: Array<Color>;
+
+}
+
+export class GraphMaterial extends MaterialBase {
+
+	constructor( parameters?: GraphMaterialParameters );
+
+	dim: boolean;
+	thickness: number;
+	graphCount: number;
+	graphDisplay: Vector4;
+	overlay: boolean;
+	xRange: Vector2;
+	yRange: Vector2;
+	colors: Array<Color>;
+	graphFunctionSnippet: string;
+
+}
+
+export type PhysicalPathTracingMaterialDefines = {
+
+	FEATURE_MIS: 0 | 1;
+	FEATURE_RUSSIAN_ROULETTE: 0 | 1;
+	RANDOM_TYPE: 0 | 1 | 2;
+
+	FEATURE_DOF: 0 | 1,
+	FEATURE_BACKGROUND_MAP: 0 | 1,
+	FEATURE_FOG: 0 | 1,
+	CAMERA_TYPE: 0 | 1 | 2,
+	DEBUG_MODE: 0 | 1,
+
+	ATTR_NORMAL: number,
+	ATTR_TANGENT: number,
+	ATTR_UV: number,
+	ATTR_COLOR: number,
+
+};
+
+export interface PhysicalPathTracingMaterialParameters extends ShaderMaterialParameters {
+
+	resolution?: Vector2;
+	bounces?: number;
+	transmissiveBounces?: number;
+	filterGlossyFactor?: number;
+
+	physicalCamera?: PhysicalCameraUniform;
+	cameraWorldMatrix?: Matrix4;
+	invProjectionMatrix?: Matrix4;
+
+	bvh?: MeshBVHUniformStruct;
+	attributesArray?: AttributesTextureArray;
+	materialIndexAttribute?: UIntVertexAttributeTexture;
+	materials?: MaterialsTexture;
+	textures?: DataArrayTexture;
+
+	// TODO: TypeScript doesn't allow overriding properties with different types
+	lights?: LightsInfoUniformStruct;
+	iesProfiles?: DataArrayTexture;
+	environmentIntensity?: number;
+	environmentRotation?: Matrix4;
+	envMapInfo?: EquirectHdrInfoUniform;
+
+	backgroundBlur?: number;
+	backgroundMap?: Texture;
+	backgroundAlpha?: number;
+	backgroundIntensity?: number;
+	backgroundRotation?: Matrix4;
+
+	seed?: number;
+	sobolTexture?: Texture;
+	stratifiedTexture?: DataTexture;
+	stratifiedOffsetTexture?: DataTexture;
+
+}
+
+export class PhysicalPathTracingMaterial extends MaterialBase<PhysicalPathTracingMaterialDefines> {
+
+	constructor( parameters?: PhysicalPathTracingMaterialParameters );
+
+	resolution: Vector2;
+	bounces: number;
+	transmissiveBounces: number;
+	filterGlossyFactor: number;
+
+	physicalCamera: PhysicalCameraUniform;
+	cameraWorldMatrix: Matrix4;
+	invProjectionMatrix: Matrix4;
+
+	bvh: MeshBVHUniformStruct;
+	attributesArray: AttributesTextureArray;
+	materialIndexAttribute: UIntVertexAttributeTexture;
+	materials: MaterialsTexture;
+	textures: DataArrayTexture;
+
+	// TODO: TypeScript doesn't allow overriding properties with different types
+	lights: LightsInfoUniformStruct;
```

Comment:
`ShaderMaterial` already uses `lights` for something else ([docs](https://threejs.org/docs/?q=r#api/en/materials/ShaderMaterial.lights)) and TypeScript doesn't allow overriding properties with a different type.

```
Property 'lights' in type 'PhysicalPathTracingMaterial' is not assignable to the same property in base type 'MaterialBase<PhysicalPathTracingMaterialDefines>'.
  Type 'LightsInfoUniformStruct' is not assignable to type 'boolean'.ts(2416)
```

Reply:
With the recent refactor I'm thinking we may not need to export this material which would simplify things quite a bit, I think.

Reply:
Even if this material may not be exported in the future, I think the TypeScript error is raising a valid concern.

Due to subtyping, any class that extends `ShaderMaterial` is expected to be compatible with its interface. Overriding `lights` with a `LightsInfoUniformStruct` may break other code that expects it to be a `boolean` as defined on `ShaderMaterial` in unexpected ways.

How do you feel about renaming this field to `lightsInfo`?